### PR TITLE
Updates Argo to latest 5.x version of Blacklight (5.18.0)

### PR DESCRIPTION
**Pull request by mejackreed**
This PR updates Argo to the latest Blacklight 5.x branch 5.18. It does continue use of the SolrDocument mixins for convenience methods. This pulls some of that logic out of the view.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -35,7 +35,7 @@ gem 'uglifier', '>= 1.0.3'
 # Stanford/Hydra related gems
 gem 'about_page'
 gem 'active-fedora'
-gem 'blacklight', '~> 5.11.3' # More deprecation warnings from 5.11.3 to 5.16
+gem 'blacklight', '~> 5' # More deprecation warnings from 5.11.3 to 5.16
```
Comment (cbeer) [inline Gemfile]: Shouldn't we pin to `~> 5.18`? Or are we ready to go to 6.x? (and, is this comment obsolete?)

File: app/models/concerns/object_type_concern.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module ObjectTypeConcern
+  extend Blacklight::Solr::Document
+
+  FIELD_OBJECT_TYPE = :objectType_ssim
+
+  ##
+  # Eventhough this is a multivalued Solr field, it should actually be treated
+  # as a single valued field.
+  def object_type
+    fetch(FIELD_OBJECT_TYPE).first
```
Comment (cbeer) [inline app/models/concerns/object_type_concern.rb]: Could we write this without exception handling (checking `has?`, maybe?)

File: app/views/catalog/_bulk_index_table.html.erb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <div id="bulk-upload-table">
     <table class="table table-hover">
 	<tr>
-	    <th>When</th><th>Who</th><th>File Name</th><th>Note</th><th>Status <%= link_to bulk_status_help_path(@document.get('id')), :data => { ajax_modal: 'trigger', ajax_modal_title: 'Status Information' } do %><span class="glyphicon glyphicon-question-sign" aria-hidden="true"></span><% end %></th><th>Druids</th><th>Success</th><th/><th/>
```
Comment (cbeer) [inline app/views/catalog/_bulk_index_table.html.erb]: Is it possible to pass just `@document` to these path helpers, or do the route not line up quite right?

Comment (mejackreed) [inline app/views/catalog/_bulk_index_table.html.erb]: Actually doesn't need a doc at all. Good catch, updating.

File: app/models/null_user.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+class NullUser
+  def is_admin?
+    false
+  end
+
+  def is_manager?
+    false
+  end
+
+  def is_viewer?
```
Comment (mejackreed) [inline app/models/null_user.rb]: Addressed now in an updated version of https://github.com/sul-dlss/argo/commit/6137adf19318612dec03e4e5ba850b65f4deddab#diff-3d4dc98d40fd4202d526f609441144dbR2

File: app/models/null_user.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+class NullUser
+  def is_admin?
+    false
+  end
+
+  def is_manager?
```
Comment (mejackreed) [inline app/models/null_user.rb]: Addressed now in an updated version of: 6137adf#diff-3d4dc98d40fd4202d526f609441144dbR2

**Comment by cbeer** [comment]
:clap:

**Pull request merged**